src/cider/nrepl/middleware/debug.clj

```diff
@@ -346,20 +350,40 @@
 (defmacro breakpoint
   "Send the result of form and its coordinates to the client.
   Sends a response to the message stored in debugger-message."
+  [form {:keys [coor]} original-form]
+  `(let [val# ~form]
+     (cond
+       (skip-breaks? ~coor) val#
+       ;; The length of `coor` is a good indicator of current code
+       ;; depth.
+       (= (first @*skip-breaks*) :trace)
+       (do (print-step-indented ~(count coor) '~original-form val#)
+           val#)
+       ;; Nothing special. Here's the actual breakpoint logic.
+       :else (->> (pr-short val#)
+                  (assoc *extras* :debug-value)
+                  (read-debug-command val#)))))
+
+(defmacro maybe-breakpoint
```

Comment:
This macro always introduces a `breakpoint`. It might be skipped during execution, but it's always there.

I prefer to call this macro `breakpoint`, and change the name of the macro above to something like `breakpoint-implementation`.